src/tmt300rs/hid-tmt300rs.c

```diff
@@ -344,6 +344,32 @@ static u8 damper_values[] = {
 	0x7f, 0x07
 };
 
+static void t300rs_calculate_periodic_values(struct ff_effect *effect)
+{
+	struct ff_periodic_effect *periodic = &effect->u.periodic;
+	int16_t headroom;
+
+	effect->replay.length -= 1;
+
+	periodic->magnitude = (periodic->magnitude * fixp_sin16(effect->direction * 360 / 0x10000)) / 0x7fff;
+
+	if (periodic->magnitude < 0){
+		/* the wheel handles positive magnitudes only */
+		periodic->magnitude = -periodic->magnitude;
+
+		/* to give the expected result 180 deg is added to the phase */
+		periodic->phase = (periodic->phase + (0x10000 / 2)) % 0x10000;
+	}
+
+	/* the interval [0; 32677[ is used by the wheel for the [0; 360[ degree phase shift */
+	periodic->phase = periodic->phase * 32677 / 0x10000;
```

Comment:
32677 is a bit surprising, I would've expected it to be 32768. I guess it could be that the driver is trying to avoid some unwanted behaviour with values at the extreme ranges?

Reply:
I think it maps to some lookup table inside the wheel, although the value choice still odd, but it works out perfectly.
Wine has some issues using the correct values through the whole range, especially from 390 to 399 deg. I don't know why it happens, but it had it before too (see spreadsheet).

Reply:
32677 + 91 = 32768, I guess the Windows driver just doesn't use the last interval or something.

> but it works out perfectly.

Does 32768 still work, or did you mean that the 32677 maximum is the highest working value?

Reply:
If we assume the Windows driver is working correctly, then `[0; 32677[ -> [0; 360[` and `[0; 65354[ -> [0; 720[`. 
The wheel responds to the whole `[0, 65535]` interval.